### ISSUE TITLE
MySQL: reporting PRS error when adding unique constraint to multiple columns

### ISSUE THREAD

**Issue by ngdah** [sqlfluff/sqlfluff#1816]
## Expected Behaviour
SQLFluff should pass lint check when adding unique constraint to multiple columns in MySQL.
Reference: https://stackoverflow.com/questions/635937/how-do-i-specify-unique-constraint-for-multiple-columns-in-mysql#answer-635943

## Observed Behaviour
SQL reported PRS error `Found unparsable section`

## Steps to Reproduce
Prepare a file add_index.sql:
```sql
ALTER TABLE `foo`.`bar` ADD UNIQUE `index_name`(`col_1`, `col_2`, `col_3`);

```
Run sqlfluff lint --dialect=mysql add_index.sql

Output will be
```
L:   1 | P:  35 |  PRS | Line 1, Position 35: Found unparsable section: '
                       | `index_name`(`col_1`, `col_2`, `col_3`)...'
All Finished 📜 🎉!
```

## Dialect
mysql

## Version
Python 3.9.7
sqlfluff, version 0.7.1

## Configuration
default

**Comment by tunetheweb** [comment]
This is a great first issue for anyone looking to start contributing to SQLFluff: https://github.com/sqlfluff/sqlfluff/wiki/Contributing-Dialect-Changes

**Comment by kayman-mk** [comment]
The same holds true for the `create table` statement.

**Comment by tunetheweb** [comment]
Raised #2443 for this.

@kayman-mk could you raise a separate issue for the `create table` issue with example SQL?

**Comment by kayman-mk** [comment]
Done with #2515
